### ISSUE TITLE
UI/UX: Size and shape category - Activate height map

### ISSUE THREAD

**Issue by dominijk** [colouring-cities/colouring-london#434]
**Dependencies** 
None 

**What UI does**
Allows input of numeric value below a maximum (550m) ~~and above a minimum (2m?)~~

**What map looks like**
As stories this is a gradated colour `#ff6161` (ignore below colours), the breaks for the colour are as below, based on analysis of LiDAR data using Jenks natural breaks in 9 divisions. @tomalrussell another mapnik question, does it support the equivalent of mapbox stops where it renders on one colour until value?

EDIT: To below values as discussed with @mz8i 

- 2.37 `#f7f4f9`
- 5.55 `#e7e1ef`
- 7.73 `#d4b9da`
- 11.38 `#c994c7`
- 18.45 `#df65b0`
- 35.05 `#e7298a`
- 89.30 `#ce1256`
- 152+ `#980043`

**Comment by tomalrussell** [comment]
Mapnik can render polygons with e.g. `3 < data_value <= 5` in the same colour - this is what we do for age, e.g. [polygon.xml#L97-L104](https://github.com/tomalrussell/colouring-london/blob/master/app/map_styles/polygon.xml#L97-L104)

There seems to be an interpolator for rendering rasters [RasterColorizer](https://github.com/mapnik/mapnik/wiki/RasterColorizer) but I can't see anything similar for vector features, if we want a smooth interpolation between stops (as you can get with vector tile data-driven styles).

**Comment by polly64** [comment]
Hi Maciej Can we activate this this week?

**Comment by mz8i** [comment]
@polly64  this one is a bit complicated, since the size and shape category already displays data based on the number of storeys (core). Before launch I had a discussion with @dominijk and understood that this feature is related to adding the ability to colour a single category based on any of the attributes (for example, use either number of storeys or height) - but that feature is a long way to go - currently we can only choose one attribute to colour by, per category.

On the other hand, if this is about switching to use height instead of storeys, then it's an easy task and can definitely be done this week.

**Comment by tomalrussell** [comment]
@mz8i I'd push this to a lower priority - after #367 as you describe

**Comment by mz8i** [comment]
@polly64 just checking that this is the task you wanted to be implemented by swapping over from number of floors to building height?

**Comment by polly64** [comment]
@dominijk @mz8i   yes thanks that's right. can u liaise with Dom on this as he is accessing height to apex data from steve at UCL Energy.  Can we also use graduated colours using colour brewer red to yellow colour ramp as storeys .  I think it's goodl to have the light buildings the highest and to have generally use lightest/closest to yellow as the least common

**Comment by dominijk** [comment]
@mz8i on the data I'll be back to this tomorrow. I've extracted it but when I tried to input via the API last week I was not getting any matches. The data uses TOID, I'm going to run a test join on the db to see if it's an issue with missing toids or something in our upload script / API/

**Comment by dominijk** [comment]
@mz8i data us being posted via the API now to production. Some data already in staging

**Comment by polly64** [comment]
Thanks Maciej

Hi Dom and maciej have had a look and think we need to change a few things and go back to the like me? yellow and red palette as much clearer and pink gradation doesn't really work very well- very hard to understand at a distance with the jenks classes as well . Also i should have said we can't use white on white background unless it has border. But it is fantastic to use as you can quickly compare similar typologies so easily as never before - amazing

Think we may need to add feet and metres as many Uk people think in feet.
I have created 7 categories below that make sense i think but we may have to experiment

200-350m ( includes for example Shard at 310 and canary wharf wharf at 235) pale yellow
100-199 (includes St Paul's 111) yellow
36-99 (includes big ben and typical 20 storey domestic towers) pale orange
20-35 orange
11-20 (Medium- large terrace including 5 storey blocks - orangy red
4-10 (small house) light red
Less than 3 (garage) dark red
also i'm not seeing all buildings coloured

**Comment by mz8i** [comment]
Further discussion on the above points:
1. Switching to different colour scale - true that pink gradation doesn't work very well in the end, yellow to red might work better. Just wondering - the usual colour scale for height maps, for example physical maps of the world, is: green - low, yellow - middle, red - tall (e.g. with the Himalayas in dark red). Do you think reversing this direction could confuse users? It might not, as the nature of the data is different (continuous height maps vs discrete buildings, plus here buildings that don't have data are dark so would blend in with the darker red if that was chosen for tall buildings) - so here the tall yellows and low reds might actually work better. Just wanted to double check what you think.
2. white on white border - is this about the lightest pink on this colour scale, or something else? If the former, agreed.
3. "But it is fantastic to use as you can quickly compare similar typologies so easily as never before - amazing" - which feature does this relate to? I don't quite understand.
4. The best we can do now is include feet in the legend text. But it would make it a bit crowded. It'd useful to have the metre values converted to approximations in feet. I don't use feet so don't have an intuition for what are good approximated value for these breaks.
5. If we're going for manual breaks, I'd assume the top one is actually anything above 200m? (to account for future high rises)

**Comment by polly64** [comment]
1. Think would be good to try red low and yellow high as its sort of graduated and discrete  and also good to have the brightest colour which is yellow to stand out as won't be many super high buildings. We might need to adjust further to make reds more distinct
2. yes sorry looked white
3. ignore just an observation of how good it is
4. Ok let's talk re feet later and just do metres for now. This is quite complicated a) as you said because it can overcrowd key but b)  because when u look at conversion tables you end up getting  things like 5ft i inch and 3/4.  Most people over 50  in Uk , apart from architects and others using metric, will probably  only know their own height  and building height in feet,

5. i just put 350 because shard is 310 i think and therefore it gives space for  a big jump still and still means people have some sort of sense of the tallest buildings.  Whatever you think best

**Comment by mz8i** [comment]
@polly64 I think it's fair to say that after we actually fixed the pink colour scale to have correct breaks (#572), it looks way better/clearer than initially. We will still evaluate the yellow-red scale, but at least it's not tragic for now

**Comment by polly64** [comment]
sorry re delays on responses will get back this afternoon


Polly Hudson

Senior Research Fellow

Director - Colouring London https://colouringlondon.org/

Bartlett Centre for Advanced Spatial Analysis

University College London

90 Tottenham Court Road W1T 4TJ



________________________________

**Comment by polly64** [comment]
@mz8i yes great for now
